zephyr/wrapper.c

```diff
@@ -178,7 +154,7 @@ void *rballoc_align(uint32_t flags, uint32_t caps, size_t bytes,
 {
 	void *ptr, *new_ptr;
 
-	bytes += PLATFORM_DCACHE_ALIGN - 1 + sizeof(void *);
+	bytes += MAX(PLATFORM_DCACHE_ALIGN - 1, sizeof(struct __alloc_hdr));
```

Comment:
This really uses the fact that sizeof(struct __aloc_hdr) is a power of 2, if it weren't this code would break. Might be worth adding a comment.

Reply:
in the code above PLATFORM_DCACHE_ALIGN is bigger then sizeof(struct __alloc_hdr), so this does not matter.
I also have build time assert here:
https://github.com/thesofproject/sof/blob/ee76c124f4b1759a85bc6c674b976411586f290d/zephyr/wrapper.c#L62

Reply:
Again, might be worth adding a comment. This assert ensures nothing breaks but should actually have a comment on why it needs to be a power of two.